packages/plugin-core/src/commands/Doctor.ts

```diff
@@ -67,6 +96,71 @@ export class DoctorCommand extends BasicCommand<CommandOpts, CommandOutput> {
         });
         break;
       }
+      case DoctorActions.CREATE_MISSING_LINKED_NOTES: {
```

Comment:
can we have the bulk of this logic in `DoctorCLICommand`? that way, it will be accessible in the CLI as well. for the CLI, we might add a `dry-run` flag in place of the preview

Reply:
Do you think it's better to make this a separate command (like `BackfillV2Command` for `DoctorActions.FIX_FRONTMATTER` does)?

or should this be a cli-first command?

Reply:
Just my two cents, but wouldn't it be better if cli, plugin, or API endpoint on the engine are just different front ends for the Command? 
Since it should be executable from the cli,  plugin, or an API call to the engine? 

The plugin could call the API to get it done, cli could execute it without needing the engine or call the engine and format the result differently for a console.

Reply:
the CLI, plugin and API endpoint are just different frontends for the command. 
the dendron engine handles the core operations (CRUD and a few extras) and it is accessed through the API.

just published an updated startup diagram [here](https://wiki.dendron.so/notes/c160ddce-edec-4f6e-841b-418d6030fa37.html)